Rendering improvements

PrototypeESBU opened the pull request:
Restructuring to remove unnecessary renders. Testing required. Incorporates PR as well #22

ronijaakkola commented:
Did some testing and it seems to work great! Didn't find anything broken.

The only edge case I encountered is that if the video panel is docked to the top, hotbar and scene navigation will be fully on top of it.

However, I think top docked video is such a rare use case that I would not account for it. Hotbar and scene navigation being (partly) on top of the video panel when it is docked to the left is better in my opinion.

PrototypeESBU commented:
Oh yeah, I see it now. The same approach can work on those as well. I'll add that in.

PrototypeESBU commented:
OK turns out it was way easier then expected. Just set the css position from fixed to absolute because those elements are are already nested within the UI containers.

I cleaned up some code and added a re-render on the scene controls if the av dock position is changed. should be good now.
